File: sic-api/Entities/Su/SuTask.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace sic_api.Entities.Su;

[Table("su_task")]
public class SuTask : BaseBusinessEntity
{
    [Required]
    [MaxLength(20)]
    [Column("task_code")]
    public string TaskCode { get; set; } = default!;

    [MaxLength(255)]
    [Column("task_name_en")]
    public string? TaskNameEn { get; set; }

    [MaxLength(255)]
    [Column("task_name_local")]
    public string? TaskNameLocal { get; set; }

    [Column("is_active")]
    public bool IsActive { get; set; } = false;

    public ICollection<SuUserTask>? UserTasks { get; set; }
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/su/SuTask.java
package com.softinter.sicapi.entity.su;

import java.util.ArrayList;
import java.util.List;

import com.softinter.sicapi.entity.base.BaseBusinessEntity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "su_task")
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class SuTask extends BaseBusinessEntity{

    @Column(name = "task_code", nullable = false, length = 20)
    private String taskCode;

    @Column(name = "task_name_en", length = 255)
    private String taskNameEn;

    @Column(name = "task_name_local", length = 255)
    private String taskNameLocal;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = false;

    @OneToMany(mappedBy = "task", fetch = FetchType.LAZY)
    private List<SuUserTask> userTasks = new ArrayList<>();
}
